client/consensus/aura/src/lib.rs

```diff
@@ -340,23 +340,23 @@ where
 		&mut self.block_import
 	}
 
-	fn epoch_data(
+	fn aux_data(
 		&self,
 		header: &B::Header,
 		_slot: Slot,
-	) -> Result<Self::EpochData, sp_consensus::Error> {
+	) -> Result<Self::AuxData, sp_consensus::Error> {
 		authorities(self.client.as_ref(), &BlockId::Hash(header.hash()))
 	}
 
-	fn authorities_len(&self, epoch_data: &Self::EpochData) -> Option<usize> {
+	fn authorities_len(&self, epoch_data: &Self::AuxData) -> Option<usize> {
```

Comment:
We don't have authorities at Subspace either, so we have hardcoded `Some(2)` here, would be great to somehow remove it from the API as well.

Reply:
Indeed currently we are using the `authorities_len()` method within the worker only to check if we are *ready* to author a block.

Currently being *ready* translates of having at least 2 authorities registered.

https://github.com/davxy/substrate/blob/b25a4fe54cb633a38972fcfde56e31440ad136e1/client/consensus/slots/src/lib.rs#L294-L298

I think that your proposal makes sense and we can rename it to `can_author` or something like that... If you like the idea please file an issue and we can eventually open a discussion about it.

Ty